src/test/java/com/conveyal/gtfs/validator/FlexValidatorTest.java

```diff
@@ -0,0 +1,563 @@
+package com.conveyal.gtfs.validator;
+
+import com.conveyal.gtfs.error.NewGTFSError;
+import com.conveyal.gtfs.error.NewGTFSErrorType;
+import com.conveyal.gtfs.model.BookingRule;
+import com.conveyal.gtfs.model.FareRule;
+import com.conveyal.gtfs.model.Location;
+import com.conveyal.gtfs.model.LocationGroup;
+import com.conveyal.gtfs.model.Stop;
+import com.conveyal.gtfs.model.StopTime;
+import com.google.common.collect.Lists;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static com.conveyal.gtfs.model.Entity.DOUBLE_MISSING;
+import static com.conveyal.gtfs.model.Entity.INT_MISSING;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class FlexValidatorTest {
+    private static final FlexValidator flexValidator = new FlexValidator();
+
+    @ParameterizedTest
+    @MethodSource("createLocationGroupChecks")
+    void validateLocationGroupTests(LocationGroupArguments locationGroupArguments) {
+        flexValidator.errors.clear();
+        flexValidator.validateLocationGroups(
+            (LocationGroup) locationGroupArguments.testObject,
+            locationGroupArguments.stops,
+            locationGroupArguments.locations
+        );
+        checkValidationErrorsMatchExpectedErrors(flexValidator.errors, locationGroupArguments.expectedErrors);
+    }
+
+    private static Stream<Arguments> createLocationGroupChecks() {
+        return Stream.of(
+            Arguments.of(
```

Comment:
You don't need to use `Arguments.of` if  only passing a single argument to each test.
(Apply to the other parametrized tests too)
```java
private static Stream<LocationGroupArguments> createLocationGroupChecks() {
    return Stream.of(
        new LocationGroupArguments(...),
        new LocationGroupArguments(...),
        ...
    );    
```